### PR TITLE
Added `increasekvi` & `decreasekvi` variables

### PR DESCRIPTION
**Pull request by ALazyMeme**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable
- [x] Documentation in docs/ or install-docs/ was updated, if applicable
- [x] I have tested all changes

This worked and is quite scuffed

### CODE REVIEW

File: docs/variables.md [hunk 1]
```diff
@@ -233,3 +233,8 @@ Example usages:
 It is not possible to add a timezone to this argument.
 
 stftime is a useful filter to call on this function
+
+#### count
+
+- `$(addcount:br_wins)` will add 1 count to the br_wins counter
+- `$(remcount:br_wins)` will remove 1 count from the br_wins counter
```
Comment (pajlada) [inline docs/variables.md:240]: Add documentation to right below `$(kvi)`

File: pajbot/bot.py [hunk 2]
```diff
@@ -477,6 +477,34 @@ def get_value(self, key, extra={}):
         log.warning("Unknown key passed to get_value: %s", key)
         return None
 
+    def add_count_value(self, key, extra={}):
```
Comment (pajlada) [inline pajbot/bot.py:480]: Rename to `increment_kvi_value` and move to below `get_kvi_value`

File: pajbot/bot.py [hunk 3]
```diff
@@ -477,6 +477,34 @@ def get_value(self, key, extra={}):
...
+            return
+
+        try:
+            self.kvi[key].inc()
+        except:
+            log.error(f'failed to add to "{key}"!')
+
+        return self.kvi[key]
+
+    def del_count_value(self, key, extra={}):
```
Comment (pajlada) [inline pajbot/bot.py:494]: Rename to `decrement_kvi_value` and move to below `get_kvi_value`

File: pajbot/models/action.py [hunk 4]
```diff
@@ -356,6 +356,8 @@ def get_substitutions(string: str, bot: Bot) -> Dict[str, Substitution]:
         method_mapping["strictargs"] = bot.get_strictargs_value
         method_mapping["command"] = bot.get_command_value
         method_mapping["broadcaster"] = bot.get_broadcaster_value
+        method_mapping["addcount"] = bot.add_count_value
+        method_mapping["delcount"] = bot.del_count_value
```
Comment (pajlada) [inline pajbot/models/action.py:360]: Rename to `increasekvi`/`decreasekvi` and move to below `kvi`

Feel free to add `inckvi` and `incrementkvi` and `deckvi` and `decrementkvi` as aliases

**Review by pajlada — CHANGES_REQUESTED** [review]

**Comment by RAnders00** [comment]
What exactly is the use case of this?

**Comment by ALazyMeme** [comment]
> What exactly is the use case of this?

Replacement of the `add_win`/`remove_win` functions, but also a count variable - which could be used for wins, literally anything that you can count.

**Review by RAnders00 — CHANGES_REQUESTED** [review]

File: docs/variables.md [hunk 5]
```diff
@@ -71,6 +71,11 @@ Customapis are generally a very powerful way of allowing pajbot to do more thing
 
 `$(kvi:active_subs)` - int - Number of subscribers to the broadcaster.
 
+#### increase/decrease kvi
+
+- `$(increasekvi:br_wins)`/`$(inckvi:br_wins)` will add 1 count to the br_wins counter
```
[the remark this answers is not in the document]
Comment (Felanbird) [reply docs/variables.md:76]: We discussed this on stream last week, and came to the conclusion that `increment` & `decrement` while correct, are not common enough words to actually get their point across. 
image


[He said something about adding a space to the variable so they can have set values to something other than 1, (idk I forgot honestly) but it was said as a response to the wording issue].

[the remark this answers is not in the document]
Comment (ALazyMeme) [reply docs/variables.md:76]: > He said something about adding a space to the variable so they can have set values to something other than 1

Said he would add it in a later pr

File: pajbot/bot.py [hunk 6]
```diff
@@ -313,6 +313,34 @@ def start(self) -> None:
     def get_kvi_value(self, key: str, extra: Dict[Any, Any] = {}) -> int:
         return self.kvi[key].get()
 
+    def increase_kvi_value(self, key, extra={}):
```
Comment (RAnders00) [inline pajbot/bot.py:316]: Could you add typing info for `key`?

File: pajbot/bot.py [hunk 7]
```diff
@@ -313,6 +313,34 @@ def start(self) -> None:
...
+            return
+
+        try:
+            self.kvi[key].inc()
+        except:
+            log.error(f'failed to add to "{key}"!')
+
+        return self.kvi[key]
+
+    def decrease_kvi_value(self, key, extra={}):
```
Comment (RAnders00) [inline pajbot/bot.py:330]: Same here

**Review by RAnders00 — CHANGES_REQUESTED** [review]

File: pajbot/bot.py (hunk 6 above)
Comment (pajlada) [reply pajbot/bot.py]: Typing + return values added to both!

**Comment by pajlada** [comment]
`$(increasekvi)` and `$(decreasekvi)` now allow you to increase or decrease the counter by a specified amount by using the syntax `$(increasekvi:KEYNAME<space>AMOUNT)` - there's some tests adding for the parsing function under `pajbot/tests/test_kvi.py`

**Review by pajlada — APPROVED** [review]
Tested this and it works as expected, documentation updated to reflect new functionality but would like one more person to check off on it ofc

**Comment by ALazyMeme** [comment]
> Tested this and it works as expected, documentation updated to reflect new functionality but would like one more person to check off on it ofc

I tried testing it, but I couldn't figure out the merging errors I was getting image
I'll leave @RAnders00 to take a look.

**Review by RAnders00 — APPROVED** [review]

**Pull request merged**